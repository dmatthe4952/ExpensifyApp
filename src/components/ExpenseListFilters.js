import React from 'react';
import { connect } from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state={
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused)=>{
        this.setState(() =>({calendarFocused}))
    }
    onTextFilterChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortByFilterChange = (e) => {
        if (e.target.value === 'date'){
            this.props.sortByDate();
        } else  if (e.target.value === 'amount'){
            this.props.sortByAmount();
        }
    }
    render(){
        return (
            <div>
                <span>
                Filter:
                <input
                    type='text'
                    value={this.props.filters.text}
                    onChange={this.onTextFilterChange}/>
                </span>
                <span>
                SortBy:
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSortByFilterChange}>
                    <option value='date' >Date</option>
                    <option value='amount' >Amount</option>
                </select>
                </span>
                <span>
                DateRange:
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={()=>{false}}
                    />
                </span>
            </div>
        )
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setStartDate: (startDate)=>(dispatch(setStartDate(startDate))),
        setEndDate: (endDate)=>(dispatch(setEndDate(endDate))),
        setTextFilter: (text)=>(dispatch(setTextFilter(text))),
        sortByDate: ()=>(dispatch(sortByDate())),
        sortByAmount: ()=>(dispatch(sortByAmount()))
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);