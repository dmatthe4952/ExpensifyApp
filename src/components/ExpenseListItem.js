import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import {startRemoveExpense} from '../actions/expenses';
import {Link} from 'react-router-dom';
import numeral from 'numeral';


export const ExpenseListItem = ({dispatch, description, id, amount, createdAt, note}) => {
    return (
        <tr>
            <td><Link to={`/edit/${id}`}>{description}</Link></td>
            <td>{numeral(amount / 100).format('$0,0.00')}</td>
            <td>{moment(createdAt).format("MM/DD/YYYY")}</td>
            <td>{note}</td>
            <td><button onClick={(e)=>{
                dispatch(startRemoveExpense({id}));
            }}>
            Remove</button></td>
        </tr>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default connect()(ExpenseListItem);