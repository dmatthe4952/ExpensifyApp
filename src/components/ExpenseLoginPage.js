import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const ExpenseLoginPage  = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__box__title" >Track My Spending</h1>
            <p>Keep your spending under control by tracking your spending habits.</p>
            <button onClick={startLogin} id="login-button" >Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps) (ExpenseLoginPage);