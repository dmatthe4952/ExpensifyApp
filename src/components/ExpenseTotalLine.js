import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseTotalLine = ({expensesCount, expensesTotal}) => { 
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
           <div className="content-container">
            {
                {expensesCount}  === 0 ?
                (<p/>) : 
                (<h1 className="page-header__title">You are viewing <span>{expensesCount}</span> {expenseWord},
                    totaling <span>{formattedExpensesTotal}</span>.</h1>)
                }
                <div>
                    <Link to="/create" className="button">Add Expense</Link>
                </div>
            </div>
        </div>
    )
 };

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseTotalLine);