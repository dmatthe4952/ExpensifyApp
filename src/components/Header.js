import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <header className="header">
        <h1>Track My Spending</h1>
        <NavLink to="/dashboard" activeClassName="is-active" exact={true}> Home </NavLink>
        <NavLink to = "/create" activeClassName="is-active"> Add an Expense </NavLink>
        <NavLink to = '/help' activeClassName='is-active'> Help </NavLink>
        <button id="logout" onClick={startLogout}  >LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
        startLogout: () =>  dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);