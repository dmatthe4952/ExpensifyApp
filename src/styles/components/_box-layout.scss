

.box-layout {
    align-items: center;
    background: url('/images/dollar-2584746_1920.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 4px;
    text-align: center;
    padding: 3.2rem, 4.6rem;
    width: 25rem;
}

.box-layout__box__title {
    margin-top: 0;
    background-color: $dark-grey;
    padding-top: 3px;
    color: #dfdfdf;
}