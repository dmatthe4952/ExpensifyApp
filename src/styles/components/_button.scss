.button {
    cursor: pointer;
    background-color: rgb(41, 106, 128);
    border-radius: 3px;
    text-decoration: none;
    color: white;
    display: inline-block;
    font-size: $l-size-font;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    border: none;
    border-radius: 5px;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}