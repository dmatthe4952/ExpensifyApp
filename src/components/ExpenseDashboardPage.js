import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseTotalLine from './ExpenseTotalLine';

const ExpenseDashboardPage = () => (
    <div className="content-container">
        <ExpenseTotalLine />
        <ExpenseListFilters />        
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;