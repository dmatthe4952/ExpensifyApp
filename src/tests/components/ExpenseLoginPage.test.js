import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseLoginPage} from '../../components/ExpenseLoginPage';

test('Should display the Login Page correctly', ()=>{
    const wrapper = shallow(<ExpenseLoginPage startLogout={()=>({})} />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout when button clicked', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<ExpenseLoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});