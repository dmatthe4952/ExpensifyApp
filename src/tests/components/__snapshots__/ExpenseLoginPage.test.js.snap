// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__box__title"
    >
      Track My Spending
    </h1>
    <p>
      Keep your spending under control by tracking your spending habits.
    </p>
    <button
      id="login-button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
